src/nupic/algorithms/Connections.cpp

```diff
@@ -330,7 +332,8 @@ Activity Connections::computeActivity(const vector<Cell>& input,
       // TODO: Possible optimization - define constant variable here?
       synapseData = dataForSynapse(synapse);
 
-      if (synapseData.permanence >= permanenceThreshold)
+      // Add epsilon to permanence before comparing to avoid rounding edge cases
+      if (synapseData.permanence + EPSILON >= permanenceThreshold)
```

Comment:
This is equivalent to:

```
synapseData.permanence >= permanenceThreshold - EPSILON
```

So the change would be equivalent to changing the `permanenceThreshold` which I don't think solves the fundamental problem, just shifts it slightly. Is that correct?